pmd-java/src/main/resources/category/java/errorprone.xml

```diff
@@ -198,6 +198,18 @@ that there are no more references to the object. It should not be invoked by app
 Note that Oracle has declared Object.finalize() as deprecated since JDK 9.
         </description>
         <priority>3</priority>
+        <properties>
+            <property name="xpath">
+                <value>
+<![CDATA[
+//MethodCall[pmd-java:matchesSig("java.lang.Object#finalize()")]
+    (: it's ok inside finalize :)
+    [not(SuperExpression and ../../../../MethodDeclaration[@Name = 'finalize'][@Arity = 0][VoidType]
+     and not(ancestor::Initializer))]
```

Comment:
I don't think we should hardcode the position of the `MethodDeclaration` with `../../...`

```suggestion
    [not(SuperExpression and ancestor::*[self::MethodDeclaration or self::Initializer][1][@Name = 'finalize'][@Arity = 0][VoidType])]
```